packages/components/src/templates/next/layouts/Collection/utils/sortCollectionItems.ts

```diff
@@ -0,0 +1,96 @@
+import type { GetCollectionItemsProps } from "./getCollectionItems"
+import type { AllCardProps } from "~/interfaces"
+
+export type SortableCardProps = AllCardProps & {
+  rawDate?: Date
+}
+
+export interface SortCollectionItemsProps
+  extends Pick<GetCollectionItemsProps, "sortBy" | "sortDirection"> {
+  items: SortableCardProps[]
+}
+
+// Sort by last updated date, tiebreaker by title
+const sortCollectionItemsByDate = ({
+  items,
+  sortDirection = "desc",
+}: Omit<SortCollectionItemsProps, "sortBy">) => {
+  return items.sort((a, b) => {
+    const bothHaveDates = a.rawDate instanceof Date && b.rawDate instanceof Date
+    const bothSameDatetime = a.rawDate?.getTime() === b.rawDate?.getTime()
+
+    if (bothHaveDates && !bothSameDatetime) {
+      // Type assertion because TS control-flow narrowing only works when
+      // check is done inline and not when we define the variable
+      const aDate = a.rawDate as unknown as Date
+      const bDate = b.rawDate as unknown as Date
+
+      switch (sortDirection) {
+        case "asc":
+          return aDate.getTime() >= bDate.getTime() ? 1 : -1
```

Comment:
nit: since we have `!bothSameDatetime` technically don't need to check for equality, but ok with this